File: 2491 Divide Players Into Teams of Equal Skill/2491.cpp
class Solution {
public:
    long long dividePlayers(vector<int>& skill) {
        sort(skill.begin(), skill.end());
        int n = skill.size();
        int a = skill[0];
        int b = skill[n-1];
        int total = a + b;
        long long ans = a * b;
        
        for (int i = 1; i < n/2; i++){
            a = skill[i];
            b = skill[n-1-i];
            if (a + b != total){
                return -1;
            } else {
                ans += a*b;
            }
        }

        return ans;
    }
};

File: 2491 Divide Players Into Teams of Equal Skill/2491.py
class Solution:
    def dividePlayers(self, skill: List[int]) -> int:
        skill.sort()
        n = len(skill)
        a, b = skill[0], skill[n-1]
        ans = a * b
        total = a + b

        for i in range(1, n//2):
            a, b = skill[i], skill[n-1-i]
            if a + b != total:
                return -1
            else: 
                ans += a * b

        return ans
